magento/pwa-studio
# [PWA-1244]: update url_suffix queries to fetch from storeConfig

## Description

Updating Components to use one source of truth for url_suffix of products and categories, fetching from storeConfig instead of product/category entities.


## Related Issue







Closes PWA-1244

## Acceptance





Refactor all queries that use category or product url_suffix, to use the storeConfig suffix.

### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature





1. Verify that user is able to navigate to a product page, and a category page when USE_STORE_CODE_IN_URL=true
2. Verify that user can switch stores and view products and category pages when USE_STORE_CODE_IN_URL=true
3. Verify that user can search for a product, and view search page results when USE_STORE_CODE_IN_URL=true
4. Verify that user can do all of the above when USE_STORE_CODE_IN_URL=false

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
